src/sdcard/mod.rs

```diff
@@ -708,12 +715,13 @@ impl Delay {
         T: embedded_hal::blocking::delay::DelayUs<u8>,
     {
         if self.retries_left == 0 {
-            Err(err)
+            return Err(err);
         } else {
```

Comment:
If the first block has a `return` in it, you don't need the else.

Reply:
Oh, yeah, that's a misrefactor. Let me fix it up quickly.